geoapi/tasks/streetview.py

```diff
@@ -0,0 +1,390 @@
+        raise StreetviewLimitException(message)
+
+    # TODO: Implement retry
+    from_tapis_to_streetview.delay(user.id,
+                                   streetview_service.id,
+                                   system_id,
+                                   path,
+                                   organization_key)
+
+def progress_error(user: User,
+                 task_uuid: UUID,
+                 status: str=None,
+                 message: str=None,
+                 logItem: Dict=None):
+  logger.error(message)
+  message = 'Error occurred'
+  NotificationsService.create(user, status, message)
+  NotificationsService.updateProgress(task_uuid=task_uuid,
+                                      status=status,
+                                      message=message,
+                                      logItem=logItem)
+
+def clean_session(streetview_instance: StreetviewInstance,
+                  user: User,
+                  task_uuid: UUID,
+                  status: str=None,
+                  message: str=None,
+                  logItem: dict=None):
+    StreetviewService.deleteInstance(streetview_instance.id)
+    progress_error(user, task_uuid, status, message, logItem)
+    remove_project_streetview_dir(user.id, task_uuid)
+
+def _from_tapis(user: User, task_uuid: UUID, systemId: str, path: str, organization_key: str):
+    client = AgaveUtils(user.jwt)
+    listing = client.listing(systemId, path)
+    files_in_directory = listing[1:]
+
+    base_filepath = get_project_streetview_dir(user.id, task_uuid)
+
+    # TODO Should handle retry
+    if not os.path.isdir(base_filepath):
+        make_project_streetview_dir(user.id, task_uuid)
+    else:
+        remove_project_streetview_dir(user.id, task_uuid)
+        make_project_streetview_dir(user.id, task_uuid)
+        NotificationsService.create(user,
+                                    "success",
+                                    "Cleaning up previous session before publish.")
+
+    img_list = []
+
+    done_files = 0
+    files_length = len(files_in_directory)
+
+    for item in files_in_directory:
+        if item.type == "dir":
+            NotificationsService.create(user,
+                                        "warning",
+                                        "Invalid upload type. Disregarding...")
+            continue
+        if item.path.suffix.lower().lstrip('.') not in features.FeaturesService.IMAGE_FILE_EXTENSIONS:
+            continue
+        try:
+            img_name = os.path.join(str(base_filepath), Path(item.path).name)
+            img_list.append(img_name)
+            client.getRawFileToPath(systemId, item.path, img_name)
+
+            done_files += 1
+
+            NotificationsService.updateProgress(task_uuid=task_uuid,
+                                                status="in_progress",
+                                                message="Collecting files from DesignSafe to geoapi",
```

Comment:
Could we drop "geoapi".  Either "Collection files" or "Collecting files from DesignSafe"